libmpv/csharp/Form1.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Globalization;
+using System.Text;
+using System.Windows.Forms;
+using System.Runtime.InteropServices;
+
+namespace mpv
+{
+    public partial class Form1 : Form
+    {
+        const int MpvFormatString = 1;
+        private IntPtr _libMpvDll;
+        private IntPtr _mpvHandle;
+
+        [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Ansi, BestFitMapping = false)]
+        internal static extern IntPtr LoadLibrary(string dllToLoad);
+
+        [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Ansi, BestFitMapping = false)]
+        internal static extern IntPtr GetProcAddress(IntPtr hModule, string procedureName);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate IntPtr MpvCreate();
+        private MpvCreate _mpvCreate;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvInitialize(IntPtr mpvHandle);
+        private MpvInitialize _mpvInitialize;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvCommand(IntPtr mpvHandle, [MarshalAs(UnmanagedType.LPArray)] string[] args);
+        private MpvCommand _mpvCommand;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvTerminateDestroy(IntPtr mpvHandle);
+        private MpvTerminateDestroy _mpvTerminateDestroy;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvSetOption(IntPtr mpvHandle, byte[] name, int format, ref uint data);
+        private MpvSetOption _mpvSetOption;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvSetOptionString(IntPtr mpvHandle, byte[] name, byte[] value);
+        private MpvSetOptionString _mpvSetOptionString;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvGetPropertystring(IntPtr mpvHandle, byte[] name, int format, ref IntPtr data);
+        private MpvGetPropertystring _mpvGetPropertyString;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvSetProperty(IntPtr mpvHandle, byte[] name, int format, ref byte[] data);
+        private MpvSetProperty _mpvSetProperty;
+
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private object GetDllType(Type type, string name)
+        {
+            IntPtr address = GetProcAddress(_libMpvDll, name);
+            if (address != IntPtr.Zero)
+                return Marshal.GetDelegateForFunctionPointer(address, type);
+            return null;
+        }
+
+        private void LoadMpvDynamic()
+        {
+            _libMpvDll = LoadLibrary("mpv-1.dll"); // The dll is included in the DEV builds by lachs0r: https://mpv.srsfckn.biz/
```

Comment:
FYI, mpv-1.dll can be P/Invoked directly. The signatures should be something like:

```c#
[DllImport("mpv-1.dll", CallingConvention = CallingConvention.Cdecl, EntryPoint = "mpv_create", ExactSpelling = true)]
internal static extern MpvCreate();
```